core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/ArtifactCoordinates.java

```diff
@@ -11,10 +11,29 @@
 
 package org.eclipse.sw360.antenna.model.artifact.facts;
 
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
 public abstract class ArtifactCoordinates<T extends ArtifactCoordinates> implements ArtifactIdentifier<T> {
     public abstract String getName();
+
     public abstract String getVersion();
 
+    public abstract String getType();
+
+    public String getPurl() {
+        return String.format("pkg:%s/%s@%s", percentEncoding(getType()), percentEncoding(getName()),
```

Comment:
The "namespace" component is still missing from here. In general a purl looks like: `pkg:<type>/<namespace>/<name>/<version>?<qualifiers>#<subpath>`, but the `?<qualifiers>#<subpath>` could be omitted for now.

If a package manager does not support a namespace, an empty string should be used, resulting in two consecutive `/` in the purl.

Reply:
but in the purl spec there are also examples like `pkg:gem/jruby-launcher@1.1.2?platform=java`

Reply:
The purl spec even explicitly says, that this segment is not allowed to be empty:

> - **namespace**:
>
>  - The optional `namespace` contains zero or more segments, separated by slash   '/'
>  - Leading and trailing slashes '/' are not significant and should be stripped    in the canonical form. They are not part of the `namespace`
>  - Each `namespace` segment must be a percent-encoded string
>  - When percent-decoded, a segment:
>
>    - must not contain a '/'
>    - **must not be empty**
>
>  - A URL host or Authority must NOT be used as a `namespace`. Use instead a    `repository_url` qualifier. Note however that for some types, the    `namespace` may look like a host.

Reply:
See discussion above, that is correct